### PR TITLE
Fixes #37433 - Add properly indented comment, to fix cloud-init yaml

### PR DESCRIPTION
**Pull request by ShimShtein**
When there is more than two spaces indentation, the yaml generated by the template will be broken.

We need to make sure the first line starts with exactly two spaces.

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
I'd like to see a test that really verifies the issue. That means rendering cloud-init with registration parameters (you don't need Katello for that since you can use host params). You probably want to special case the test to verify the result is valid YAML.

**Comment by sayan3296** [comment]
I confirm that all issues are fixed with the revert commit i.e. https://github.com/theforeman/foreman/pull/10289/files and then the working indent commit i.e. https://github.com/theforeman/foreman/pull/10295/files . I could successfully deploy PXE and Cloud-init systems on VMware for both RHEL 8.10 and 9.4, without any issues.

( i tested with foreman 3.9 which is aligned with Sat 6.15.3 but yeah, the fix works as it should be )

File: app/services/foreman/template_snapshot_service.rb [hunk 1]
```diff
@@ -70,6 +70,7 @@ def define_host_params(host)
         "syspurpose_usage" => "Development/Test",
         "syspurpose_sla" => "Self-Support",
         "syspurpose_addons" => "first addon, second addon, third addon",
+        "subscription_manager" => "true",
```
Comment (ekohl) [inline app/services/foreman/template_snapshot_service.rb:73]: Perhaps I missed it, but for better coverage we should implement this in develop too.

**Review by ekohl — APPROVED** [review]

File: app/services/foreman/template_snapshot_service.rb (hunk 1 above)
Comment (ShimShtein) [reply app/services/foreman/template_snapshot_service.rb:73]: > Perhaps I missed it, but for better coverage we should implement this in develop too.

Already in: https://github.com/theforeman/foreman/blob/cd1ea006eb8b4bc7910e86cf2d182b1b7b63c45f/app/services/foreman/template_snapshot_service.rb#L78

**Pull request merged**